maven/plan.ps1

```diff
@@ -9,10 +9,11 @@ $pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
 $pkg_source="http://mirrors.gigenet.com/apache/${pkg_name}/maven-3/${pkg_version}/binaries/apache-maven-${pkg_version}-bin.zip"
 $pkg_shasum="7e6cfe98dc9c16ae6aa267db277860594695144d719c99d1fc519e89346a8edf"
 $pkg_dirname="apache-$pkg_name-$pkg_version"
+$pkg_bin_dirs=@('bin')
 
-function Invoke-SetupEnvironment {
-    Set-RuntimeEnv MAVEN_HOME "$pkg_prefix" [-force]
- }
+function Invoke-Prepare {
+    $env:JAVA_HOME = "$(hab pkg path core/corretto8)"
```

Comment:
I believe this will still fail. Did you mean to use `Get-HabPackagePath`?

Reply:
@echohack is correct.  We should use `Get-HabPackagePath` on Windows or `pkg_path_for` on Linux.  The reason for this is that `hab pkg path` will always return the _latest on disk_, which is potentially unsafe.  If you were to have a newer version on disk than what your Plan expects, you would end up using that rather than the intended version. Additionally, if you had a package installed, but forgot to declare it as a dependency, then the next time someone built they may end up with a failed build. 

`Get-HabPackagePath` and `pkg_path_for` only consider the resolved fully qualified identifiers (origin/name/version/release) so are safe to use in order to resolve paths.

Reply:
I believe I have corrected the issues,.  I refactored the corretto8 plan as well to make sure both are good.